pkg/capture/capture.go

```diff
@@ -223,14 +233,59 @@ func (c *Capture) process(ctx context.Context) <-chan error {
 			// This logic may be slightly more contrived than a select{} statement but it increases
 			// packet throughput by several percent points
 			if len(c.capLock.request) > 0 {
-				<-c.capLock.request             // Consume the request
+				<-c.capLock.request             // Consume the lock request
 				c.capLock.confirm <- struct{}{} // Confirm that process() is not processing
-				<-c.capLock.done                // Consume the request to continue normal processing
+
+				// Continue fetching packets and add them to the local buffer
```

Comment:
This is much cleaner now than the initial attempt. It's still a bit of duplication, but due to the fact that several `continue` and `return` cases are involved in a nested loop it's going to be hard to take this apart / factor it out into sub-functions without increasing complexity (and probably degrading performance).